commons/commons-transpiler/src/main/kotlin/ac/uk/ebi/transpiler/cli/TranspilerCommandLine.kt

```diff
@@ -0,0 +1,56 @@
+package ac.uk.ebi.transpiler.cli
+
+import ac.uk.ebi.biostd.SubFormat
+import ac.uk.ebi.transpiler.service.FilesTableTemplateTranspiler
+import org.apache.commons.cli.DefaultParser
+import org.apache.commons.cli.HelpFormatter
+import org.apache.commons.cli.Options
+import java.io.File
+
+const val BASE = "base"
+const val COLUMNS = "columns"
+const val DIR = "directory"
+const val FORMAT = "format"
+const val TEMPLATE = "template"
+
+class TranspilerCommandLine(
+    private val helpFormatter: HelpFormatter = HelpFormatter(),
+    private val transpiler: FilesTableTemplateTranspiler = FilesTableTemplateTranspiler()
+) {
+    fun transpile(args: Array<String>): String {
+        var pageTab = ""
+        val options = Options().apply {
+            addOption("b", BASE, true, "Base that will be used as prefix for the generated files path")
+            addOption("c", COLUMNS, true, "Comma separated list of columns that map to the directory structure")
+            addOption("d", DIR, true, "Path to the directory containing the files")
+            addOption("f", FORMAT, true, "Desired format for the generated page tab: TSV, JSON or XML")
+            addOption("t", TEMPLATE, true, "Path to the file containing the template")
+        }
+
+        try {
+            val cmd = DefaultParser().parse(options, args)
+            val base = cmd.getOptionValue(BASE)!!
+            val dir = cmd.getOptionValue(DIR)!!
+            val format = cmd.getOptionValue(FORMAT)!!
+            val columns = cmd.getOptionValue(COLUMNS)!!
+            val template = File(cmd.getOptionValue(TEMPLATE)!!)
+
+            pageTab = transpiler.transpile(
+                template = template.readText(),
+                baseColumns = columns.split(",").map { it.trim() }.toList(),
+                filesPath = dir,
+                basePath = base,
+                format = SubFormat.valueOf(format)
+            )
+        } catch (exception: Exception) {
+            when (exception) {
+                is NullPointerException -> println("ERROR: All the arguments are required")
```

Comment:
so if an argument is mising `NullPointerException` is throw? it surprice me that library trow such generic error and not information of even the parameter that was mising considering you are flaging parameters as required

Reply:
You were absolutely right man, the library does have that option, it's just really hidden in the documentation 😅